### ISSUE TITLE
More onConnecting fallout

### ISSUE THREAD

**Issue by oberstet** [crossbario/autobahn-python#1181]
More fallout from `onConnecting` - in `tox -e pytest` of crossbarfx:

```
test/functests/cbtests/test_cb_ssl.py::test_verification Generating a 2048 bit RSA private key
..............................................................................................................+++
........+++
writing new private key to '/tmp/tmpl0msksjk/private.key'
-----
2019-04-19T21:33:44+0200 start_crossbar
2019-04-19T21:33:44+0200 
2019-04-19T21:33:44+0200    running in /tmp/cts_authp6kpmpwk

2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177] 
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177]     :::::::::::::::::
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177]           :::::          _____                      __
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177]     :::::   :   :::::   / ___/____ ___   ___  ___  / /  ___ _ ____
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177]     :::::::   :::::::  / /__ / __// _ \ (_-< (_-< / _ \/ _ `// __/
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177]     :::::   :   :::::  \___//_/   \___//___//___//_.__/\_,_//_/
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177]           :::::
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177]     :::::::::::::::::   Crossbar v19.4.1
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177] 
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177]     Copyright (c) 2013-2019 Crossbar.io Technologies GmbH, licensed under AGPL 3.0.
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177] 
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177] Initializing <crossbar.node.node.Node> as node [realm=crossbar, cbdir=/tmp/cts_authp6kpmpwk]
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177] New node key pair generated! Public key is 0xefe33450412be54d21a487a304b55b97671edd2e49ca676ae966470416343611
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177] File permissions on node private key fixed
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177] Node key loaded from /tmp/cts_authp6kpmpwk/key.priv
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177] Node configuration loaded [config_source=localfile, config_path=/tmp/cts_authp6kpmpwk/config.json]
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177] Entering event reactor ...
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177] Starting standalone node <crossbar.node.node.Node.start>
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177] Node ID intel-nuci7 set from hostname
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177] No extra node router roles
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177] RouterServiceAgent ready (realm_name="crossbar", on_ready=None)
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177] Registered 23 procedures
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177] Signal handler installed on process 5177 thread 140066019184768
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177] Using default node shutdown triggers ['shutdown_on_worker_exit']
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177] Booting node <crossbar.node.node.Node.boot>
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177] Booting node from local configuration [parallel_worker_start=False] .. <crossbar.node.node.Node.boot_from_config>
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177] Will start 1 worker ..
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177] Order node to start Router ssl_testee
2019-04-19T21:33:45+0200 2019-04-19T21:33:45+0200 [Controller   5177] Starting router worker ssl_testee <crossbar.node.controller.NodeController.start_worker>

2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Router       5190] Starting worker "ssl_testee" for node "intel-nuci7" on realm "crossbar" with personality "standalone" <crossbar.worker.router.RouterController>
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Router       5190] Running as PID 5190 on CPython-EPollReactor
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Router       5190] Entering event reactor ...
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Router       5190] Router worker session for "ssl_testee" joined realm "crossbar" on node router <crossbar.worker.router.RouterController.onJoin>
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Router       5190] Registered 41 procedures
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Router       5190] Router worker session for "ssl_testee" ready
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Controller   5177] Ok, node has started Router ssl_testee
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Controller   5177] Configuring Router ssl_testee ..
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Controller   5177] Order Router ssl_testee to start Realm realm001
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Router       5190] Starting router realm realm001 <crossbar.worker.router.RouterController.start_router_realm>
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Router       5190] RouterServiceAgent ready (realm_name="auth_realm", on_ready=<Deferred at 0x7f21bb42c4a8>)
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Router       5190] Realm "realm001" (name="auth_realm") started
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Controller   5177] Ok, Router ssl_testee has started Realm realm001
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Controller   5177] Order Realm realm001 to start Role role001
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Router       5190] Starting role "role001" on realm "realm001" <crossbar.worker.router.RouterController.start_router_realm_role>
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Router       5190] role role001 on realm realm001 started
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Controller   5177] Ok, Realm realm001 has started Role role001
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Controller   5177] Order Realm realm001 to start Role role002
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Router       5190] Starting role "role002" on realm "realm001" <crossbar.worker.router.RouterController.start_router_realm_role>
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Router       5190] role role002 on realm realm001 started
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Controller   5177] Ok, Realm realm001 has started Role role002
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Controller   5177] Order Router ssl_testee to start Transport test_ssl_0
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Router       5190] Starting router transport "test_ssl_0" <crossbar.worker.router.RouterController.start_router_transport>
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Router       5190] Creating router transport for "test_ssl_0" <crossbar.worker.transport.create_router_transport>
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Router       5190] Router transport created for "test_ssl_0" <crossbar.worker.transport.RouterWebTransport>
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Router       5190] Created "static" Web service on root path "/" of Web transport "test_ssl_0"
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Router       5190] Loading server TLS key from /tmp/tmpl0msksjk/private.key
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Router       5190] Loading server TLS certificate from /tmp/tmpl0msksjk/server.crt
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Router       5190] Using secure default TLS ciphers
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Router       5190] No OpenSSL DH parameter file set - DH cipher modes will be deactive!
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Router       5190] Site (TLS) starting on 6464
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Controller   5177] Ok, Router ssl_testee has started Transport test_ssl_0
2019-04-19T21:33:46+0200 Detected transport starting up
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Controller   5177] Order Transport test_ssl_0 to start Web Service webservice001
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Router       5190] Starting "websocket" Web service on path "tls_ws" of transport "test_ssl_0" <crossbar.worker.router.RouterController.start_web_transport_service>
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Controller   5177] Ok, Transport test_ssl_0 has started Web Service webservice001
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Controller   5177] Ok, Router ssl_testee configured
2019-04-19T21:33:46+0200 2019-04-19T21:33:46+0200 [Controller   5177] Ok, local node configuration booted successfully!
2019-04-19T21:33:46+0200 Unhandled Error
Traceback (most recent call last):
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/twisted/python/log.py", line 103, in callWithLogger
    return callWithContext({"system": lp}, func, *args, **kw)
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/twisted/python/log.py", line 86, in callWithContext
    return context.call({ILogContext: newCtx}, func, *args, **kw)
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/twisted/python/context.py", line 122, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/twisted/python/context.py", line 85, in callWithContext
    return func(*args,**kw)
--- <exception caught here> ---
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/twisted/internet/posixbase.py", line 619, in _doReadOrWrite
    why = selectable.doWrite()
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/twisted/internet/tcp.py", line 627, in doConnect
    self._connectDone()
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/twisted/internet/tcp.py", line 656, in _connectDone
    self.protocol.makeConnection(self)
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/twisted/protocols/tls.py", line 213, in makeConnection
    ProtocolWrapper.makeConnection(self, transport)
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/twisted/protocols/policies.py", line 75, in makeConnection
    self.wrappedProtocol.makeConnection(self)
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/twisted/internet/protocol.py", line 514, in makeConnection
    self.connectionMade()
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/twisted/internet/endpoints.py", line 117, in connectionMade
    self._wrappedProtocol.makeConnection(self.transport)
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/twisted/internet/protocol.py", line 514, in makeConnection
    self.connectionMade()
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/autobahn/twisted/websocket.py", line 97, in connectionMade
    self._connectionMade()
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/autobahn/websocket/protocol.py", line 3374, in _connectionMade
    self.startHandshake()
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/autobahn/websocket/protocol.py", line 3492, in startHandshake
    transport_details = self._create_transport_details()
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/autobahn/twisted/websocket.py", line 255, in _create_transport_details
    channel_id_type=u"tls-unique",
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/autobahn/twisted/util.py", line 126, in transport_channel_id
    m.update(tls_finished_msg)
builtins.TypeError: object supporting the buffer API required

FAILED
test/functests/cbtests/test_cb_ssl.py::test_verification_fails 2019-04-19T21:33:51+0200 connecting once using transport type "websocket" over endpoint "tcp"
2019-04-19T21:33:51+0200 Starting factory <autobahn.twisted.websocket.WampWebSocketClientFactory object at 0x7f8e9d860fd0>
2019-04-19T21:33:51+0200 Unhandled Error
Traceback (most recent call last):
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/twisted/python/log.py", line 103, in callWithLogger
    return callWithContext({"system": lp}, func, *args, **kw)
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/twisted/python/log.py", line 86, in callWithContext
    return context.call({ILogContext: newCtx}, func, *args, **kw)
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/twisted/python/context.py", line 122, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/twisted/python/context.py", line 85, in callWithContext
    return func(*args,**kw)
--- <exception caught here> ---
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/twisted/internet/posixbase.py", line 619, in _doReadOrWrite
    why = selectable.doWrite()
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/twisted/internet/tcp.py", line 627, in doConnect
    self._connectDone()
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/twisted/internet/tcp.py", line 656, in _connectDone
    self.protocol.makeConnection(self)
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/twisted/protocols/tls.py", line 213, in makeConnection
    ProtocolWrapper.makeConnection(self, transport)
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/twisted/protocols/policies.py", line 75, in makeConnection
    self.wrappedProtocol.makeConnection(self)
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/twisted/internet/protocol.py", line 514, in makeConnection
    self.connectionMade()
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/twisted/internet/endpoints.py", line 117, in connectionMade
    self._wrappedProtocol.makeConnection(self.transport)
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/twisted/internet/protocol.py", line 514, in makeConnection
    self.connectionMade()
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/autobahn/twisted/websocket.py", line 97, in connectionMade
    self._connectionMade()
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/autobahn/websocket/protocol.py", line 3374, in _connectionMade
    self.startHandshake()
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/autobahn/websocket/protocol.py", line 3492, in startHandshake
    transport_details = self._create_transport_details()
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/autobahn/twisted/websocket.py", line 255, in _create_transport_details
    channel_id_type=u"tls-unique",
  File "/home/oberstet/scm/crossbario/crossbarfx/.tox/functests-cb/lib/python3.7/site-packages/autobahn/twisted/util.py", line 126, in transport_channel_id
    m.update(tls_finished_msg)
builtins.TypeError: object supporting the buffer API required

2019-04-19T21:33:51+0200 Stopping factory <autobahn.twisted.websocket.WampWebSocketClientFactory object at 0x7f8e9d860fd0>
```

**Comment by meejah** [comment]
Oh, so I think this might be from the "is it secure" detection; Twisted docs say you only need to check `ISSLTransport` ... but maybe that's not entirely/always true?

**Comment by meejah** [comment]
Okay, I think I see the problem -- just trying to confirm: it's always including channel-id in the `TransportDetails` but it should only do that if "secure" is true.

**Comment by oberstet** [comment]
maybe fixed via https://github.com/crossbario/autobahn-python/pull/1179

**Comment by oberstet** [comment]
no, it's not fixed. the issue is still here. I wanted to work on sth else and now jiggle around with this - which I actually don't care about (onConnecting) as we dont need it=(

**Comment by oberstet** [comment]
ok, I back out `onConnecting` via https://github.com/crossbario/autobahn-python/pull/1183

**Comment by meejah** [comment]
see https://github.com/crossbario/autobahn-python/pull/1185

**Comment by oberstet** [comment]
ok, merged above PR -- but crossbarfx CI runs red (pls see my comments on the PR .. sorry, first commented there, should have commented here)

**Comment by meejah** [comment]
https://github.com/crossbario/crossbarfx/pull/289 fixes part of the problem ("`pip 19.1` broken) and https://github.com/crossbario/crossbarfx/pull/290 fixes the other part (sphinx install resulted in version conflicts)

**Comment by meejah** [comment]
Okay, those two are merged and CI is green on trunk for fx so closing this.
